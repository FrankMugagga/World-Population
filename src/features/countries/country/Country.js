import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import { getCountry } from './country.Slice';
import { getContinents } from '../continentsSlice';
import '../../../style/country.css';

const Country = () => {
  const { continent } = useParams();
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCountry(continent));
    dispatch(getContinents());
  }, [dispatch, continent]);
  const countryData = useSelector((state) => state.countries.country);
  const contine = useSelector((state) => state.continents.continent);

  const [searchQuery, setSearchQuery] = useState('');
  const filteredCountries = countryData.filter(
    (country) => country.name.official.toLowerCase().includes(searchQuery.toLocaleLowerCase()),
  );

  return (
    <div className="fullCont">
      <div className="continentsCont">
        <header className="CountriesHeader">
          <div className="backKey">
            <Link className="direct" to="/"><img className="direction_k" alt="back arrow" src="https://svgsilh.com/svg/97591-ffffff.svg" /></Link>
            <div className="year">2018</div>
          </div>
          <div className="contHead">
            {continent}
            &apos;s Population
          </div>
          <div className="setM">
            <img className="microphone" alt="microphone" src="https://svgsilh.com/svg/468291-ffffff.svg" />
            <img className="settings" alt="settings" src="https://svgsilh.com/svg/98446-ffffff.svg" />
          </div>
        </header>
        <h2>
          {contine && contine.map((contin) => (
            <div key={contin.name}>
              {contin.name === continent && (
              <p className="imgCont">
                <img className="contImage" src={contin.mapUrl} alt={`${contin.name} map`} />
                <div className="popCont">
                  Total Population
                  <br />
                  {contin.population}
                </div>
              </p>
              )}
            </div>
          ))}
        </h2>
        <div className="stat">
          Country break down - 2018&nbsp;
          <input
            name="searchQuery"
            className="searchQuery"
            placeholder="search by country"
            type="search"
            value={searchQuery}
            onChange={(e) => setSearchQuery(e.target.value)}
          />
        </div>
        <section className="">
          <ul className="breakdown">
            {filteredCountries.map((country) => (
              <li className="countryList" key={country.capital}>
                <div className="countryDet">
                  <div className="forwardIcon">
                    <Link className="directionCont" to={`/countryDetail/${country.cca2}`}>
                      <img className="direction_k" alt="direction key" src="https://svgsilh.com/svg/24826-ffffff.svg" />
                    </Link>
                  </div>
                  <div className="couflag">{country.flag}</div>
                  <div className="countNam">{country.name.official}</div>
                  <div className="cpop">{country.population}</div>
                </div>

              </li>
            ))}
          </ul>
        </section>

      </div>

    </div>

  );
};

export default Country;
