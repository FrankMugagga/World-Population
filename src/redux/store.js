import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from '../features/countries/countriesSlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,

  },

});

export default store;
