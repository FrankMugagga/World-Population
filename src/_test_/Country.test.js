import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import store from '../redux/store';
import Country from '../features/countries/country/Country';

describe('performing test for Country component', () => {
  it('testing if Counties page really matches', () => {
    const continentRender = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Country />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(continentRender).toMatchSnapshot();
  });

  test('That jest works', () => {
    expect(true).toBe(true);
  });
});
