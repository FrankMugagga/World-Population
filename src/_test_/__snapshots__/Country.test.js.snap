// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performing test for Country component testing if Counties page really matches 1`] = `
<div
  className="fullCont"
>
  <div
    className="continentsCont"
  >
    <header
      className="CountriesHeader"
    >
      <div
        className="backKey"
      >
        <a
          className="direct"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="back arrow"
            className="direction_k"
            src="https://svgsilh.com/svg/97591-ffffff.svg"
          />
        </a>
        <div
          className="year"
        >
          2018
        </div>
      </div>
      <div
        className="contHead"
      >
        's Population
      </div>
      <div
        className="setM"
      >
        <img
          alt="microphone"
          className="microphone"
          src="https://svgsilh.com/svg/468291-ffffff.svg"
        />
        <img
          alt="settings"
          className="settings"
          src="https://svgsilh.com/svg/98446-ffffff.svg"
        />
      </div>
    </header>
    <h2 />
    <div
      className="stat"
    >
      COUNTRY BREAKDOWN - 2018 
      <input
        className="searchQuery"
        name="searchQuery"
        onChange={[Function]}
        placeholder="search by country"
        type="search"
        value=""
      />
    </div>
    <section
      className=""
    >
      <ul
        className="breakdown"
      />
    </section>
  </div>
</div>
`;

exports[`performing test for Country component testing if Counties page really mtches 1`] = `
<div
  className="fullCont"
>
  <div
    className="continentsCont"
  >
    <header
      className="CountriesHeader"
    >
      <div
        className="backKey"
      >
        <a
          className="direct"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="back arrow"
            className="direction_k"
            src="https://svgsilh.com/svg/97591-ffffff.svg"
          />
        </a>
        <div
          className="year"
        >
          2018
        </div>
      </div>
      <div
        className="contHead"
      >
        's Population
      </div>
      <div
        className="setM"
      >
        <img
          alt="microphone"
          className="microphone"
          src="https://svgsilh.com/svg/468291-ffffff.svg"
        />
        <img
          alt="settings"
          className="settings"
          src="https://svgsilh.com/svg/98446-ffffff.svg"
        />
      </div>
    </header>
    <h2 />
    <div
      className="stat"
    >
      COUNTRY BREAKDOWN - 2018 
      <input
        className="searchQuery"
        name="searchQuery"
        onChange={[Function]}
        placeholder="search by country"
        type="search"
        value=""
      />
    </div>
    <section
      className=""
    >
      <ul
        className="breakdown"
      />
    </section>
  </div>
</div>
`;
